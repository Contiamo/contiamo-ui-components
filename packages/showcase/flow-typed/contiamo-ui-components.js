declare module "contiamo-ui-components" {
  declare var exports: any
}
