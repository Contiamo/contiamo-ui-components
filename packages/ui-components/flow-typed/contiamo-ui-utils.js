declare module "contiamo-ui-utils" {
  declare var exports: any
}
