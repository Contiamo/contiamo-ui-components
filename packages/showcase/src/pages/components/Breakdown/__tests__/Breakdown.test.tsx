import * as React from "react"
import { render } from "enzyme"

import { wrapTheme } from "contiamo-ui-utils"
import { contiamoTheme } from "contiamo-ui-components"
import ThemelessBreakdown from "../Breakdown"

const Breakdown = wrapTheme(contiamoTheme)(ThemelessBreakdown)

describe("Breakdown Page", () => {
  it("Should render correctly", () => {
    expect(render(<Breakdown />)).toMatchSnapshot()
  })
})
