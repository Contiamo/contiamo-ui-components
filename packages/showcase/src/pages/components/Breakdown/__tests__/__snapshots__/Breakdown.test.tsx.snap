// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breakdown Page Should render correctly 1`] = `
<div
  class="css-1dkvbfz"
>
  <div
    class="css-imab3e"
  >
    Breakdown
  </div>
  <p>
    Breakdowns are a means of representing aggregated data in a way that should be relatively easy to reason about. The breakdown component itself belongs within the context of a larger container component that calculates numbers and supplies them to said component.
  </p>
  <h4>
    Usage
  </h4>
  <div
    class="css-1r6pn67"
  >
    <div
      class="css-10kxzv6"
    >
      Give yourself some space - expand this playground
    </div>
    <div
      class="playground"
    >
      <div
        class="playgroundCode"
      >
        <div
          class="react-codemirror2 playgroundStage"
        />
      </div>
      <div
        class="playgroundPreview"
      >
        <div>
          <div
            class="previewArea"
          />
        </div>
      </div>
    </div>
  </div>
  <h4>
    Props
  </h4>
  <table
    class="css-xa6ngs"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Default
        </th>
        <th>
          Type
        </th>
        <th>
          Optional?
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          number
        </td>
        <td>
          A number by which the breakdown is represented.
        </td>
        <td>
          -
        </td>
        <td>
          number
        </td>
        <td>
          ðŸš«
        </td>
      </tr>
      <tr>
        <td>
          label
        </td>
        <td>
          A statistic number label within the bar of the breakdown
        </td>
        <td>
          -
        </td>
        <td>
          string
        </td>
        <td>
          ðŸš«
        </td>
      </tr>
      <tr>
        <td>
          fill
        </td>
        <td>
          The percentage to fill the bar. This is typically passed in from a container component that calculates percentages at large.
        </td>
        <td>
          -
        </td>
        <td>
          number
        </td>
        <td>
          ðŸš«
        </td>
      </tr>
      <tr>
        <td>
          color
        </td>
        <td>
          A theme palette color name, or a hex code that the bar will be colored with.
        </td>
        <td>
          *info*
        </td>
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          icon
        </td>
        <td>
          An icon that is displayed on the breakdown
        </td>
        <td>
          -
        </td>
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          onMouseEnter/onMouseLeave
        </td>
        <td>
          Functions that are invoked when the mouse enters and/or leaves the breakdown. Useful for tooltips/infowindows
        </td>
        <td>
          -
        </td>
        <td>
          func
        </td>
        <td>
          âœ…
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
